docs/snippets/ov_common.cpp

```diff
@@ -74,10 +74,35 @@ int main() {
     //! [ov_api_2_0:create_infer_request]
 
     inputs_aligned(infer_request);
+
+    //! [ov_api_2_0:set_callback]
+    infer_request.set_callback([&] (std::exception_ptr exception_ptr) {
+        if (exception_ptr) {
+            // procces exception or rethrow it.
+            std::rethrow_exception(exception_ptr);
+        } else {
+            // Extract inference result
+            ov::Tensor output_tensor = infer_request.get_output_tensor();
+            // Restart inference if needed
+            // infer_request.start_async();
+        }
+    });
+    //! [ov_api_2_0:set_callback]
+
     //! [ov_api_2_0:inference]
     infer_request.infer();
     //! [ov_api_2_0:inference]
 
+    //! [ov_api_2_0:start_async_and_wait]
+    // Start inference without blocking current thread
+    infer_request.start_async();
+    // Get inference status
+    if (!infer_request.wait_for(std::chrono::milliseconds{0})) {
```

Comment:
Why not just `wait()` without check status?

Reply:
just to represent the difference between different timed wait API

Reply:
agree that it can confuse users since it's migration guide - the steps should be clear without confusions. To demonstrate infer request API - update https://github.com/openvinotoolkit/openvino/blob/master/docs/OV_Runtime_UG/ov_infer_request.md

Reply:
Simplified